aspnetcore/mvc/models/file-uploads.md

```diff
@@ -206,8 +206,20 @@ For a `files` input element to support uploading multiple files provide the `mul
 
 The individual files uploaded to the server can be accessed through [Model Binding](xref:mvc/models/model-binding) using <xref:Microsoft.AspNetCore.Http.IFormFile>. The sample app demonstrates multiple buffered file uploads for database and physical storage scenarios.
 
+<a name="filename"></a>
+
 > [!WARNING]
-> Don't rely on or trust the `FileName` property of <xref:Microsoft.AspNetCore.Http.IFormFile> without validation. The `FileName` property should only be used for display purposes and only after HTML encoding the value.
+> Do **not** use the `FileName` property of <xref:Microsoft.AspNetCore.Http.IFormFile> other than for display and logging. When displaying or logging, HTML encode the file name. An attacker can provide a malicious filename, including full paths or relative paths. Applications should:
+>
+> * Strip the path from the user supplied filename.
+> * Generate a filename.
+> * Save the HTML encoded stripped filename for UI or logging
```

Comment:
```suggestion
> * Save the HTML-encoded, stripped filename for UI or logging.
```

Here, too ... this might need to add something for the file name used when downloading.

Reply:
btw -- "stripped" seems unclear ... out of context, since nothing was said about stripping characters.